features/fields.feature

```diff
@@ -171,84 +201,129 @@ Functionaliteit: Fields
       | burgerservicenummer | <burgerservicenummer> |
-      En de persoon heeft de volgende 'verblijfplaats' gegevens
-      | naam                             | waarde                |
-      | straatnaam (11.10)               | <straat>              |
-      | huisnummer (11.20)               | <huisnummer>          |
-      | postcode (11.60)                 | <postcode>            |
-      | locatiebeschrijving (12.10)      | <locatiebeschrijving> |
-      | land.code                        | <land>                |
-      | regel 1 adres buitenland (13.30) | <regel1>              |
-      | regel 2 adres buitenland (13.40) | <regel2>              |
-      | regel 3 adres buitenland (13.50) |                       |
+      En heeft de persoon alleen de volgende 'geboorte' gegevens
+      | naam              | waarde         |
+      | datum.type        | <type>         |
+      | datum.datum       | <datum>        |
+      | datum.jaar        | <jaar>         |
+      | datum.maand       | <maand>        |
+      | datum.onbekend    | <onbekend>     |
+      | datum.langFormaat | <lang formaat> |
+
+      Voorbeelden:
+      | burgerservicenummer | type           | datum      | jaar | maand | onbekend | lang formaat     |
+      | 999991929           | Datum          | 1956-11-15 |      |       |          | 15 november 1956 |
+      | 999992351           | JaarMaandDatum |            | 1978 | 3     |          | maart 1978       |
+      | 999992806           | JaarDatum      |            | 1968 |       |          | 1968             |
+      | 999994220           | DatumOnbekend  |            |      |       | true     | onbekend         |
+
+    Abstract Scenario: Gevraagd veld is een gegeven van een andere type van de polymorfe gegevensgroep dan het ontvangen type
       Als personen wordt gezocht met de volgende parameters
-      | naam                | waarde                              |
-      | type                | RaadpleegMetBurgerservicenummer     |
-      | burgerservicenummer | <burgerservicenummer>               |
-      | fields              | verblijfplaats.verblijfadres.regel3 |
-      Dan heeft de response een persoon met alleen de volgende 'verblijfplaats' gegevens
-      | naam                 | waarde   |
-      | type                 | <type>   |
+      | naam                | waarde                                   |
+      | type                | RaadpleegMetBurgerservicenummer          |
+      | burgerservicenummer | <burgerservicenummer>                    |
+      | fields              | burgerservicenummer,geboorte.datum.datum |
+      Dan heeft de response een persoon met alleen de volgende gegevens
+      | naam                | waarde                |
+      | burgerservicenummer | <burgerservicenummer> |
+      En heeft de persoon alleen de volgende 'geboorte' gegevens
+      | naam           | waarde     |
+      | datum.type     | <type>     |
+      | datum.onbekend | <onbekend> |

```

Comment:
Hoe past het wel leveren van de datum bij de regel "wanneer het gegeven geen waarde heeft (ook niet onbekend) wordt de gegevensgroep niet geleverd en dus ook 'type' niet teruggegeven"

Het lastige hierbij is dat voor een datum we eigenlijk met "gegeven" de hele datum bedoelen, en niet het gevraagde veld binnen de datum. Een datum is dus een heel ander polymorfe groep dan de andere polymorfe groepen (verblijfplaats en nationaliteit) die elk uit meerdere gegevens bestaan.

Waarom is het voorbeeld van verblijfplaats verwijderd?

Ik denk dat er nu geen regel meer is die uitlegt dat het bij dit type required veld "onbekend" geleverd wordt (heb je hieronder verwijderd). Of lees ik iets verkeerd?

Reply:
Ik zie nu dat er nog meer scenario's zijn verwijderd. Dit is niet bewust maar komt omdat ik de fields feature gebruik en aanpas tijdens het ontwikkelen. Daardoor moet ik stashen om mijn aanpassingen veilig te stellen als ik develop wil mergen. Na de merge van develop, merge ik mijn gestashte aanpassingen. Als ik geen merge conflict krijg, dan ga ik er van uit dat er geen aanpassingen vanuit develop zijn verwijderd, wat dus wel het geval is

Reply:
Het is niet handig om in een feature file met een Achtergrond, Gegeven stappen toe te voegen in een scenario. Door automation code leidt dit tot meerdere personen met dezelfde bsn. Daarom zijn de Gegeven stappen verplaatst naar de Achtergrond